c2h/generators.cu

```diff
@@ -413,17 +437,37 @@ void init_key_segments(const c2h::device_vector<OffsetT>& segment_offsets, KeyT*
   auto d_range_dsts  = thrust::make_transform_iterator(d_offsets, dst_transform_op);
   auto d_range_sizes = thrust::make_transform_iterator(iota, offset_to_size_t<OffsetT>{d_offsets});
 
+#if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
   std::uint8_t* d_temp_storage   = nullptr;
   std::size_t temp_storage_bytes = 0;
+  // TODO(bgruber): replace by a non-CUB implementation
   cub::DeviceCopy::Batched(
     d_temp_storage, temp_storage_bytes, d_range_srcs, d_range_dsts, d_range_sizes, total_segments);
 
   c2h::device_vector<std::uint8_t> temp_storage(temp_storage_bytes);
   d_temp_storage = thrust::raw_pointer_cast(temp_storage.data());
 
+  // TODO(bgruber): replace by a non-CUB implementation
   cub::DeviceCopy::Batched(
     d_temp_storage, temp_storage_bytes, d_range_srcs, d_range_dsts, d_range_sizes, total_segments);
   cudaDeviceSynchronize();
+#else // THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
+  static_assert(sizeof(OffsetT) == 0, "Need to implement a non-CUB version of cub::DeviceCopy::Batched");
+  // TODO(bgruber): implement and *test* a non-CUB version, here is a sketch:
```

Comment:
I will provide and test this alternative when I integrate C2H into Thrust's unit tests. Now, I cannot drive this code path, since C2H is only used from CUB currently.